### PR TITLE
fix(multi-mcap): add minimum cache floor per source to prevent crash

### PR DESCRIPTION
**Pull request by luluiz**
## User-Facing Changes
 
 Fix crash when opening 300+ remote MCAP files. The application no longer throws
 "Requested more data than cache size" during initialization of large multi-file sessions.
 
 ## Description
 
 When loading multiple remote MCAPs, the 500 MiB total cache budget was split equally
 across all sources (`totalCache / n`). With 300+ files this produced per-source
 allocations smaller than a single MCAP metadata read (summary section, chunk indexes),
 causing `CachedFilelike` to throw.
 
 **Fix:** apply a 10 MiB minimum floor per source using `Math.max(floor, totalCache / n)`
 so every source always has enough cache for metadata reads regardless of how many files
 are opened. When the floor causes the effective total to exceed the configured budget,
 a `log.warn` is emitted.
 
 ## Checklist
 
 - [x] The web version was tested and it is running ok
 - [x] The desktop version was tested and it is running ok
 - [x] This change is covered by unit tests
 - [x] Files constants.ts, types.ts and *.style.ts have been checked and relevant code snippets have been relocated

## Summary by CodeRabbit

* **New Features**
  * Added an optional per-source minimum cache size setting for multi-remote loading.
  * Improved how total cache is distributed across remote sources using the new minimum floor.
  * Emits a warning if the enforced minimum per-source cache allocation exceeds the configured budget.
* **Tests**
  * Expanded coverage for cache-allocation edge cases, including default and custom minimum floor behavior, and verified warning emission counts.

### CODE REVIEW

File: packages/suite-base/src/players/IterablePlayer/shared/MultiIterableSource.ts [hunk 1]
```diff
@@ -54,10 +63,22 @@ export class MultiIterableSource<T extends ISerializedIterableSource, P>
         (file) => new this.SourceConstructor({ type: "file", file } as P),
       );
     } else {
-      // Distribute total cache budget evenly across remote sources.
-      // Default total budget: 500MiB (same as single-file default).
+      // Distribute total cache budget across remote sources with a minimum floor per source.
+      // A pure linear split (totalCache / n) can produce a per-source budget smaller than a
+      // single MCAP metadata read when n > ~300, causing a crash in CachedFilelike.
       const totalCache = this.dataSource.totalCacheSizeInBytes ?? 1024 * 1024 * 500;
```
Comment (rodrigo-p-rodrigues) [inline packages/suite-base/src/players/IterablePlayer/shared/MultiIterableSource.ts]: please create a const with the default cache total

```suggestion
      const totalCache = this.dataSource.totalCacheSizeInBytes ?? DEFAULT_CACHE_TOTAL_BYTES;
```

**Review by rodrigo-rodrigues-ctw — CHANGES_REQUESTED** [review]
One thing that could improve this further is somehow having a `minCacheSize` value in metadata that could guide us a bit better, and then we could have these 10MB as fallback.
With some kind of server side mcap pre-processing this would be easy to introduce.

File: packages/suite-base/src/players/IterablePlayer/shared/MultiIterableSource.ts (hunk 1 above)
Comment (luluiz) [reply packages/suite-base/src/players/IterablePlayer/shared/MultiIterableSource.ts]: done

**Review by gabriela-almeida-ctw — APPROVED** [review]

**Review by rodrigo-rodrigues-ctw — APPROVED** [review]
